au4/src/effects/effects_base/internal/effectsactionscontroller.cpp

```diff
@@ -22,5 +33,27 @@ void EffectsActionsController::doEffect(const muse::actions::ActionData& args)
 
     muse::String effectId = args.arg<muse::String>(0);
 
-    effectExecutionScenarion()->performEffect(effectId);
+    effectExecutionScenario()->performEffect(effectId);
+}
+
+void EffectsActionsController::repeatLastEffect(const muse::actions::ActionData&)
+{
+    IF_ASSERT_FAILED(m_lastProcessorId.has_value()) {
+        return;
+    }
+    effectExecutionScenario()->repeatEffect(*m_lastProcessorId);
+}
+
+bool EffectsActionsController::canReceiveAction(const muse::actions::ActionCode& code) const
+{
+    if (code == "repeat-last-effect") {
+        return m_lastProcessorId.has_value();
+    } else {
+        return true;
+    }
+}
+
+muse::async::Notification EffectsActionsController::lastEffectIsNowAvailable() const
+{
+    return m_lastEffectIsNowAvailable;
```

Comment:
There is no need to add this notification to the controller, for the same reasons described here https://github.com/audacity/audacity/pull/7368/files#r1768242195